src/DnsClient/NameServer.cs

```diff
@@ -95,6 +96,7 @@ public NameServer(IPAddress endPoint, int port, string dnsSuffix = null)
         public NameServer(IPEndPoint endPoint, string dnsSuffix = null)
         {
             IPEndPoint = endPoint ?? throw new ArgumentNullException(nameof(endPoint));
+            DnsSuffix = string.IsNullOrWhiteSpace(dnsSuffix) ? null : dnsSuffix;
```

Comment:
Ups, good catch, actually an error I added yesterday.
Could you change the ctor above, too, and pass dnsSuffix through instead of setting it there?